auth/user-store.ts

```diff
@@ -1,27 +1,69 @@
-import { User, UserSchema } from "./user";
+import { User, UserSchema, UserInfo } from "./user";
+import { SocialSignInMock } from "./social-signin-mock";
 
 export class UserStore {
   private nextId = 0;
-  private idStore = new Map<string, UserSchema>();
-  private emailStore = new Map<string, UserSchema>();
+  private store = new Map<string, UserSchema>();
+  private readonly socialMocks: SocialSignInMock[] = [];
```

Comment:
Maybe it would make sense for this to be a `Map` or `Set` to simplify `consumeSocialMock()` with methods like `.get()`, `.remove()` or `.has()`?